pkg/proto/types_test.go

```diff
@@ -2102,3 +2104,110 @@ func TestStateHashDebutUnmarshalJSON(t *testing.T) {
 		assert.Equal(t, sh, shd.GetStateHash())
 	}
 }
+
+func TestAttachmentInOrder(t *testing.T) {
+	const (
+		seed        = "3TUPTbbpiM5UmZDhMmzdsKKNgMvyHwZQncKWfJrxk3bc"
+		matcher     = "7kPFrHDiGw1rCm7LPszuECwWYL3dMf6iMifLRDJQZMzy"
+		amountAsset = "8LQW8f7P5d5PZM7GtZEBgaqRPGSzS3DfPuiXrURJ4AJS"
+		priceAsset  = "2bkjzFqTMM3cQpbgGYKE8r7J73SrXFH8YfxFBRBterLt"
+	)
+	s, err := base58.Decode(seed)
+	require.NoError(t, err)
+	_, pk, err := crypto.GenerateKeyPair(s)
+	require.NoError(t, err)
+	mpk := crypto.MustPublicKeyFromBase58(matcher)
+	aa, err := NewOptionalAssetFromString(amountAsset)
+	require.NoError(t, err)
+	pa, err := NewOptionalAssetFromString(priceAsset)
+	require.NoError(t, err)
+	ts := uint64(time.Now().UnixMilli())
+	oV4 := OrderV4{
+		Version:         4,
+		MatcherFeeAsset: OptionalAsset{},
+		PriceMode:       OrderPriceModeDefault,
+		OrderBody: OrderBody{
+			SenderPK:  pk,
+			MatcherPK: mpk,
+			AssetPair: AssetPair{
+				AmountAsset: *aa,
+				PriceAsset:  *pa,
+			},
+			OrderType:  Sell,
+			Price:      uint64(100),
+			Amount:     uint64(1000),
+			Timestamp:  ts,
+			Expiration: ts + 100*1000,
+			MatcherFee: uint64(10),
+		},
+	}
+	oV4.Attachment = make([]byte, 0)
+	ok, err := oV4.Valid()
+	assert.NoError(t, err)
+	assert.True(t, ok)
+
+	oV4.Attachment = make([]byte, MaxAttachmentSize)
+	ok, err = oV4.Valid()
+	assert.NoError(t, err)
+	assert.True(t, ok)
+
+	oV4.Attachment = make([]byte, MaxAttachmentSize+1)
+	ok, err = oV4.Valid()
+	assert.EqualError(t, err, "attachment size should be <= 1024 bytes, got 1025")
+	assert.False(t, ok)
```

Comment:
On my mind creating different test cases with new order instances would be cleaner than this.